servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/AsyncContext.java

```diff
@@ -68,11 +76,42 @@ static AsyncContextProvider provider() {
      * Get the current {@link AsyncContextMap}.
      *
      * @return the current {@link AsyncContextMap}
+     * @deprecated Use {@link #context()}.
      */
+    @Deprecated
     public static AsyncContextMap current() {
         return provider().contextMap();
     }
 
+    /**
+     * Get the current {@link ContextMap}.
+     *
+     * @return the current {@link ContextMap}
+     */
+    public static ContextMap context() {
+        return provider().context();
+    }
+
+    /**
+     * Registers a new mapping between {@link AsyncContextMap.Key} and {@link ContextMap.Key} to allow smooth migration
+     * to the new API.
+     * <p>
+     * If the mapping is registered using this method, it will be possible to use both keys for {@link AsyncContext}
+     * operations at the same time, allowing step-by-step migration from deprecated to the new keys. For example, it
+     * will be possible to use {@link #put(AsyncContextMap.Key, Object)} and {@link #get(ContextMap.Key)} to retrieve
+     * the same value.
+     *
+     * @param acmKey {@link AsyncContextMap.Key} to map
+     * @param cmKey {@link ContextMap.Key} to map
+     * @param <T> type of the value for these keys
+     * @deprecated This method can be used only as a temporary approach until the full codebase is migrated to
+     * {@link ContextMap} API. This method will be removed in future releases.
+     */
+    @Deprecated
+    public static <T> void newKeyMapping(final AsyncContextMap.Key<T> acmKey, final ContextMap.Key<T> cmKey) {
```

Comment:
> Have you considered having the registration happen implicitly with a possible static utility to enable/disable the behaviour?

We give a chance to users to register a mapping upfront. If nothing was registered, we always have to do that implicitly. Disabling that implicit mapping will break old API users because internal implementation is already migrated to the new API.


> Registering a mapping at `Key` creation time might not be too much overhead

Hm... This sounds like a good idea performance-wise. However, if I register all mappings upfront implicitly (at `newKey` time) and let users override mappings through this public utility later, they will lose visibility into those `info` level logs in `AsyncContextMapToContextMapAdapter` that help to identify which keys still use deprecated API.  WDYT?

Reply:
Got it. I actually made that remark before seeing the implicit registrations :) Regarding the optimization and info logs about deprecated API -> it's a tradeoff -> perhaps we should measure what performance hit will typical workloads take with and without the logs and decide based on that data.

Reply:
This mapping is required only for 0.41.x and will be removed in 0.42. We can prioritize optimization if someone reports a visible impact. In practice, users don't have too many keys created at runtime. The hit should happen only for the first use of a key.

Reply:
Removed in #1959